deluge/httpdownloader.py

```diff
@@ -152,7 +154,11 @@ def request_callback(self, response):
                     self.filename = new_file_name
 
             cont_type_header = headers.getRawHeaders(b'content-type')[0].decode()
-            cont_type, params = cgi.parse_header(cont_type_header)
+            message = email.message.Message()
+            message['content-type'] = cont_type_header
+            message_params = message.get_params()
```

Comment:
We can simplify with 
```
cont_type = message.get_content_type()
params = message['content-type'].params
```

Reply:
Thank you . Actually https://docs.python.org/3/library/cgi.html#cgi.parse_header says your way is preferred. Will update.